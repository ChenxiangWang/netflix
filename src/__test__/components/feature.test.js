import React from "react";
import { render, screen } from "@testing-library/react";
import Feature from "../../components/feature";

describe("<Feature/> Test", () => {
  it("Render test", () => {
    const { container } = render(
      <Feature>
        <Feature.Title>test-tittle</Feature.Title>
        <Feature.SubTitle>test-subtitle</Feature.SubTitle>
      </Feature>
    );
    const titleEl = screen.getByText("test-tittle");
    const subTitleEl = screen.getByText("test-subtitle");
    expect(titleEl).toBeTruthy();
    expect(subTitleEl).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
