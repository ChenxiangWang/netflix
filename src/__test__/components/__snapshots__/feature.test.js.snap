// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature/> Test Render test 1`] = `
<div
  class="style__Container-sc-n1qvws-0 bkxYDa"
>
  <h1
    class="style__Title-sc-n1qvws-1 iSAhNJ"
  >
    test-tittle
  </h1>
  <h2
    class="style__SubTitle-sc-n1qvws-2 jhNzDN"
  >
    test-subtitle
  </h2>
</div>
`;
